src/common/lib/api.ts

```diff
@@ -70,26 +71,37 @@ export const swrGetAccountInfo = async (
   const accountsCache = await getAccountsCache();
 
   return new Promise((resolve, reject) => {
-    if (accountsCache[id]) {
+    // don't apply cache for connectors that failed to connect
+    if (accountsCache[id] && !accountsCache[id].error) {
       if (callback) callback(accountsCache[id]);
       resolve(accountsCache[id]);
     }
 
     // Update account info with most recent data, save to cache.
     getAccountInfo()
       .then((response) => {
-        const { alias } = response.info;
-        const { balance: resBalance, currency } = response.balance;
-        const name = response.name;
-        const balance =
-          typeof resBalance === "number" ? resBalance : parseInt(resBalance); // TODO: handle amounts
-        const account = {
-          id,
-          name,
-          alias,
-          balance,
-          currency: currency || "BTC", // set default currency for every account
-        };
+        let account: AccountInfo;
+        if (response.error) {
+          account = {
```

Comment:
Shouldn't this case be handled in the catch() block here? 

msg.request should throw an Error():

https://github.com/lujakob/lightning-browser-extension/blob/feat/not-reachable-error/src/common/lib/msg.ts#L19-L21

Reply:
Your suggestion was my first approach. The problem is, the argument of `throw new Error()` is of type string. But I need to pass an object, to be able to get the `name` of the connector and `error`. To change this, I need to refactor dozens of places where `msg.request` is called in a try/catch block. So on the one hand this is really a big refactor, on the other hand I think what we do now is not completely wrong: we return a connector in error state, that has `name` as its only property. Considering these two points I opted for the current approach. (Maybe it doesn't feel so wrong for me, because I used this approach in other applications, where the error returns data and therefore is not dealt with as a real error) What do you think?